lib/mayaHydra/mayaPlugin/renderOverride.cpp

```diff
@@ -1436,6 +1447,10 @@ void MtohRenderOverride::_CreateSceneIndicesChainAfterMergingSceneIndex(const MH
             Fvp::DisplayStyleOverrideSceneIndex::New(_inputSceneIndexOfFilteringSceneIndicesChain);
     _displayStyleSceneIndex->addExcludedSceneRoot(MAYA_NATIVE_ROOT); // Maya native prims don't use global refinement
 
+    // Add texture disabling Scene Index
+    _lastFilteringSceneIndexBeforeCustomFiltering = _pruneTexturesSceneIndex = 
+    Fvp::PruneTexturesSceneIndex::New(_lastFilteringSceneIndexBeforeCustomFiltering);
```

Comment:
I think you should create the pruneTextureSceneIndex only when the texture are off in the viewport. And not put it in the scene index chain when textures are on.
Always creating this scene index and having it not doing much when textures are on is a performance concern. 
It will always still do things with the empty lambda or be an indirection for GetPrim and will slow down the pipeline.

Reply:
I had it the same way earlier, but that meant the entire scene prims get removed and added on every toggle. This could be very expensive. Currently, I am only dirtying the prims on toggle which I think is less expensive. The method MarkTexturesDirty() is invoked on toggling. Also, the scene index is created only once during initialization now. I have made changes in _NeedToRecreateTheSceneIndicesChain() where texture toggling condition is not accounted for anymore. Perhaps we could handle this more elegantly when we have a new mechanism via FVP API to handle dirtying.

Reply:
The fact that the entire scene prims are removed and added again is logged in a separate defect. Let's talk about this offline.